### PR TITLE
Restore Quick Chat CORS compatibility control

### PR DESCRIPTION
**Pull request by zeroliu**
Fixes logancyang/obsidian-copilot-preview#313

## Why

The v4 BYOK migration stopped carrying the legacy `enableCors` choice into the new provider model. Connection verification still succeeded because it uses Obsidian's CORS-free request path, but Quick Chat then constructed the model without that choice and used browser `fetch`, so endpoints without browser CORS headers failed with a connection error. Endpoint construction itself did not change.

## What

| Area | Behavior |
| --- | --- |
| Provider settings | Restores an explicit **Enable CORS** switch with an information icon and a fixed-width tooltip that explains the Quick Chat streaming tradeoff. |
| Quick Chat | Persists the provider choice and carries it into the legacy chat runtime. Enabled uses Obsidian's buffered CORS-free path; disabled keeps token streaming. |
| Migration | Future legacy BYOK migrations preserve `enableCors`. Models sharing an endpoint but carrying different choices migrate into separate providers so neither choice is lost. |
| Agent Mode | A CORS-only edit does not restart OpenCode, and `enableCors` is not forwarded into OpenCode's provider config. |
| Provenance | Requires future behavioral branches and edge-case tests to retain the full originating issue URL in both code rationale and test description. |

## Non goal

- Retrofit providers for users whose one-time migration already completed; they can set the restored switch manually.
- Change provider Base URL construction or connection verification.
- Change OpenCode or other Agent Mode network transports.

## Screenshot

**Before**

Provider dialog before restoring the CORS control

**After**

Provider dialog with Enable CORS and its Quick Chat tooltip

## Risk

**High**

| Criterion | Status | Reason |
| --- | --- | --- |
| No behavior change, or a cosmetic/copy/docs/config change visible where it renders, or deterministic tests cover the changed behavior | ❌ | This intentionally restores a persisted transport choice and changes Quick Chat request behavior when enabled. |
| A defect would fail CI or be obvious on first use | ✅ | The setting, bridge, migration, and Agent isolation paths have deterministic coverage and the control was exercised in Obsidian. |
| A revert fully restores prior state, including persisted data | ❌ | New provider rows can retain the optional `enableCors` field after a revert, although older code ignores it. |
| No auth, permissions, secrets, or input-handling surface changes | ✅ | Credentials and endpoint inputs are unchanged. |
| No public API, plugin API, message, or on-disk contract changes | ❌ | The persisted provider contract gains an optional `enableCors` field. |
| No core-path concurrency, async-lifecycle, or state-machine changes | ✅ | Request construction remains synchronous; the only notification change suppresses an unnecessary Agent restart for CORS-only edits. |
| No hot-path behavior lacks deterministic coverage | ✅ | Quick Chat bridging, provider persistence, migration, and OpenCode config isolation are covered. |
| No new dependency | ✅ | Dependency manifests are unchanged. |
| Human-only behavior stays in one feature area and surfaces quickly | ✅ | The visual and streaming behavior is confined to the provider dialog and the first Quick Chat response. |

Review: inspect `ConfigureProviderForm`, `configuredModelToCustomModel()`, `ProviderRegistry.update()`, `planByokMigration()`, and `buildOpencodeConfig()`, then run Verification steps 1–5.

## Verification

1. Edit a BYOK provider, hover the information icon beside **Enable CORS**, and confirm the fixed-width tooltip names Quick Chat and explains that enabled responses arrive after completion instead of token by token.
2. Turn **Enable CORS** on, save, reopen the provider, and confirm the choice remains on. Send a Quick Chat request through an endpoint without browser CORS headers and confirm the completed response appears.
3. Turn **Enable CORS** off for an endpoint that supports browser CORS and confirm Quick Chat streams token by token.
4. Run a fresh legacy migration fixture containing both enabled and disabled CORS choices for the same endpoint; confirm both choices survive in separate providers, while a matching provider that already exists is left unchanged.
5. Start an OpenCode Agent session, change only **Enable CORS**, and confirm the session is not restarted and the generated provider config does not contain `enableCors`.

## Note

The original private work endpoint was not available for a live reproduction. The same transport boundary was verified through the model bridge tests, and the new UI plus persisted setting was exercised against an existing configured provider in the test vault.

### CODE REVIEW

**Comment by zeroliu** [comment]
@codex review

File: src/modelManagement/chatModel/configuredModelToCustomModel.ts [hunk 1]
```diff
@@ -144,6 +144,16 @@ export function configuredModelToCustomModel(params: {
...
     baseUrl: provider.baseUrl,
     apiKey: resolvedApiKey,
+    // https://github.com/logancyang/obsidian-copilot-preview/issues/313:
+    // custom endpoints can pass verification through requestUrl but fail chat
+    // when native fetch enforces browser CORS. Catalog-backed providers keep
+    // native fetch so their responses can stream incrementally.
+    enableCors:
+      provider.providerType === "openai-compatible" &&
+      provider.origin.kind === "byok" &&
+      provider.origin.catalogProviderId === undefined
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/modelManagement/chatModel/configuredModelToCustomModel.ts]: Fixed in 54a37c11 by restoring an explicit provider-level **Enable CORS** choice instead of inferring from catalog linkage. That covers catalog providers pointed at custom proxies as well as catalog-less endpoints. The setting is persisted, bridged only into Quick Chat, and covered by the configured-model bridge and provider-dialog tests.

**Comment by zeroliu** [comment]
@codex review
